src/dune_lang/blang.ml

```diff
@@ -15,6 +15,20 @@ module Ast = struct
   let true_ = Const true
   let false_ = Const false
 
+  let rec equal f t1 t2 =
+    match t1, t2 with
+    | Const b1, Const b2 -> Bool.equal b1 b2
+    | Not t1, Not t2 -> equal f t1 t2
+    | Expr x1, Expr x2 -> f x1 x2
+    | And tl1, And tl2 | Or tl1, Or tl2 ->
+      (match List.for_all2 ~f:(equal f) tl1 tl2 with
```

Comment:
Isn't `List.equal` good here?

Reply:
Yes, it is. Amended as suggested, thanks.